webroot/src/pages/FinalizePrivilegePurchase/FinalizePrivilegePurchase.ts

```diff
@@ -0,0 +1,522 @@
+            if (stateSelected?.fee) {
+                total += stateSelected.fee;
+            }
+
+            if (stateSelected?.isMilitaryDiscountActive && stateSelected?.militaryDiscountAmount) {
+                total -= stateSelected.militaryDiscountAmount;
+            }
+        });
+
+        return total;
+    }
+
+    get isSubmitEnabled(): boolean {
+        this.checkValidForAll();
+
+        return this.isFormValid && this.formData.noRefunds.value && !this.isFormLoading;
+    }
+
+    //
+    // Methods
+    //
+    async handleSubmit() {
+        this.validateAll({ asTouched: true });
+
+        if (this.isFormValid) {
+            this.startFormLoading();
+            this.isFormError = false;
+            this.formErrorMessage = '';
+            const { formData, statesSelected } = this;
+
+            const serverData = LicenseeUserPurchaseSerializer.toServer({ formData, statesSelected });
+
+            const purchaseServerEvent = await this.$store.dispatch('user/postPrivilegePurchases', serverData);
+
+            this.endFormLoading();
+
+            if (purchaseServerEvent?.message === 'Successfully processed charge') {
+                this.$router.push({
+                    name: 'PurchaseSuccessful',
+                    params: { compact: this.currentCompactType }
+                });
+            } else if (purchaseServerEvent?.message) {
+                this.isFormError = true;
+                this.formErrorMessage = purchaseServerEvent?.message;
+            }
+        } else {
+            this.isFormError = true;
+            this.formErrorMessage = this.formValidationErrorMessage;
+        }
+    }
+
+    initFormInputs() {
+        this.formData = reactive({
+            firstName: new FormInput({
+                id: 'first-name',
+                name: 'first-name',
+                label: this.firstNameInputLabel,
+                shouldHideLabel: false,
+                shouldHideMargin: true,
+                autocomplete: 'given-name',
+                placeholder: this.firstNamePlaceHolderText,
+                validation: Joi.string().required().messages(this.joiMessages.string),
+                value: '',
+            }),
+            lastName: new FormInput({
+                id: 'last-name',
+                name: 'last-name',
+                label: this.lastNameInputLabel,
+                shouldHideLabel: false,
+                shouldHideMargin: true,
+                autocomplete: 'family-name',
+                placeholder: this.lastNamePlaceHolderText,
+                validation: Joi.string().required().messages(this.joiMessages.string),
+                value: '',
+            }),
+            expMonth: new FormInput({
+                id: 'exp-month',
+                name: 'exp-month',
+                label: '',
+                shouldHideLabel: true,
+                shouldHideMargin: true,
+                placeholder: '00',
+                autocomplete: 'cc-exp-month',
```

Comment:
Im getting "Automatic credit card filling is disabled because this form does not use a secure connection."

From googling this it seems like it wont autofill on http. Im going to ngrok it and see if I can I confirm it works